lib/internal/crypto/diffiehellman.js

```diff
@@ -232,8 +236,37 @@ function getFormat(format) {
   return POINT_CONVERSION_UNCOMPRESSED;
 }
 
+function diffieHellman(ourKey, theirKey) {
+  if (!(ourKey instanceof KeyObject))
+    throw new ERR_INVALID_ARG_TYPE('ourKey', 'KeyObject', ourKey);
+
+  if (!(theirKey instanceof KeyObject))
+    throw new ERR_INVALID_ARG_TYPE('theirKey', 'KeyObject', theirKey);
+
+  if (ourKey.type !== 'private')
+    throw new ERR_CRYPTO_INVALID_KEY_OBJECT_TYPE(ourKey.type, 'private');
+
+  if (theirKey.type !== 'public' && theirKey.type !== 'private') {
+    throw new ERR_CRYPTO_INVALID_KEY_OBJECT_TYPE(theirKey.type,
+                                                 'private or public');
+  }
+
+  const ourType = ourKey.asymmetricKeyType;
+  const theirType = theirKey.asymmetricKeyType;
+  if ((ourType !== 'dh' || theirType !== 'dh') &&
```

Comment:
Could this be `if (ourType ~== theirType) throw ...; if(!['dh', 'ed', ..].includes(ourType)) throw ...`? Would be a bit more readable. Or perhaps the key object itself can have a property to indicate it is "dh capable", to avoid the list of algs?

Reply:
I used a set instead, hoping that it might be faster (might actually be slower for such a small number of key types). Thanks!